core/src/model.jl

```diff
@@ -129,17 +133,26 @@ function Model(config::Config)::Model
     tstops = sort(unique(vcat(tstops...)))
     adaptive, dt = convert_dt(config.solver.dt)
 
-    jac_prototype = config.solver.sparse ? get_jac_prototype(parameters) : nothing
+    jac_prototype =
+        config.solver.sparse ? get_jac_prototype(parameters, length(u0)) : nothing
     RHS = ODEFunction(water_balance!; jac_prototype)
 
     @timeit_debug to "Setup ODEProblem" begin
         prob = ODEProblem(RHS, u0, timespan, parameters)
     end
     @debug "Setup ODEProblem."
 
-    callback, saved = create_callbacks(parameters, config, saveat)
+    callback, saved = create_callbacks(parameters, config, u0, saveat)
     @debug "Created callbacks."
 
+    # Only have finite tolerance on storage states
+    abstol = copy(u0)
+    reltol = copy(u0)
+    abstol .= Inf
+    reltol .= Inf
+    abstol.storage .= config.solver.abstol
+    reltol.storage .= config.solver.reltol
+
```

Comment:
If desired, we could apply different tolerances to different components of the states this way